### PR TITLE
Link Pipeline concepts page to relevant concept docs

### LINKED ISSUE #1624: Link Pipeline concepts page to relevant concept docs
Page: https://www.kubeflow.org/docs/pipelines/overview/concepts/pipeline/

The Pipeline concept page relies on a number of terms that are defined in other concept docs. Add links to these docs for the first instance of: 
- Component
- Step

Also, it would be great to update the last sentence to specify that pipelines can be uploaded using the UI or the Kubeflow Pipelines SDK.

### PR DESCRIPTION
**Pull request by janeman98**
Fixes #1624

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1657)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
/assign @joeliedtke 
/unassign

I've assigned @joeliedtke as reviewer/approver, as he's the lead tech writer for Pipelines.

File: content/docs/pipelines/overview/concepts/pipeline.md [hunk 1]
```diff
@@ -5,17 +5,16 @@ weight = 10
...
 A *pipeline* is a description of a machine learning (ML) workflow, including all
-of the components in the workflow and how the components relate to each other in
+of the [components](/docs/pipelines/overview/concepts/component/) in the workflow and how the components relate to each other in
 the form of a [graph](/docs/pipelines/concepts/graph/). The pipeline
 configuration includes the definition of the inputs (parameters) required to run
 the pipeline and the inputs and outputs of each component.
 
 When you run a pipeline, the system launches one or more Kubernetes Pods
-corresponding to the steps (components) in your workflow (pipeline). The Pods
+corresponding to the [steps (components)](/docs/pipelines/overview/concepts/step/) in your workflow (pipeline). The Pods
```
Comment (joeliedtke) [inline content/docs/pipelines/overview/concepts/pipeline.md]: Suggestion: Would you please move (components) outside of the link for steps?

```
[steps](/docs/pipelines/overview/concepts/step/) (components)
```

File: content/docs/pipelines/overview/concepts/pipeline.md [hunk 2]
```diff
@@ -5,17 +5,16 @@ weight = 10
...
 start Docker containers, and the containers in turn start your programs.
 
-After developing your pipeline, you can upload and share it on the 
-Kubeflow Pipelines UI.
+After developing your pipeline, you can upload using the Kubeflow Pipelines UI or the Kubeflow Pipelines SDK.
```
Comment (joeliedtke) [inline content/docs/pipelines/overview/concepts/pipeline.md]: Suggested revision: 

After developing your pipeline, you can upload your pipeline using the Kubeflow Pipelines UI or the Kubeflow Pipelines SDK.

File: content/docs/pipelines/overview/concepts/pipeline.md (hunk 1 above)
Comment (janeman98) [reply content/docs/pipelines/overview/concepts/pipeline.md]: Thanks @joeliedtke for reviewing this PR. I have modified the file as what you suggested. Please review/approve. Feel free to let me know if further change is needed.

File: content/docs/pipelines/overview/concepts/pipeline.md (hunk 2 above)
Comment (janeman98) [reply content/docs/pipelines/overview/concepts/pipeline.md]: Thanks @joeliedtke for reviewing this PR. I have modified the file as what you suggested.

**Comment by joeliedtke** [comment]
/lgtm
/approve

Thank you for working on this!

**Comment by janeman98** [comment]
Many thanks, @joeliedtke !

**Pull request merged**